### ISSUE TITLE
[9.0 Enterprise] No possibility to print statutary reports (PL & BS) before retained earnings

### ISSUE THREAD

**Issue by fclementic2c** [odoo/odoo#8820]
With Odoo 9 no more opening period and no more automatic opening journal entry.
The purpose of this opening entry was to bring forward account balances AND post profit/loss of previous fiscal year to the retained earnings account (cumulating earning over years minus dividends).
In V9, reports will compute initial balances on the fly from the very 1st entry posted in Odoo...fine but what about the retained earning account ? 
I guess accountants will simply post manually the profit of the current year to the retained earning.
The issue is that if I print the P&L after this entry the profit/loss will always be zero (because transfered to retained earning account)
It is usually no big deal for accounting softwares beacause there is always a possibility to print the P&L and B/S before and after the retained earning entry. 
However, in Odoo v9 there is no such feature.

**Comment by fclementic2c** [comment]
@qdp-odoo Would you mind to have a look at this one please ? thxs

**Comment by qdp-odoo** [comment]
i'm not sur i got your point... you say that once you posted the current year earnings your P&L will always be 0? 

Let me explain how you're suppose to do and probably you'll spot where you made an error because according to me it's working just fine ;-)
Let say that your fiscal year 2014 ended on the 31th of December. During 2015, you didn't had to do anything to have a correct P&L because Odoo was filtering accounting entries from the 1st of January. Your BS was correct to because the "current year entries" line was including the result of 2014 too (which allowed you to work on both 2014 and 2015 on the same time without having to struggle with opening entries). Right? 
Ok, let's say that today you wanna post the 2014 year earnings in the right equity accounts: you check your P&L for year 2014 (in the filter) and spread the amount into your accounts, BUT for the counterpart(s) instead of using your accounts from the P&L to zero them out, you use the special account "Undistributed profit/loss" which is of type "current year earnings" and has been created automatically by the system for you. That way, if you wanna print the P&L for 2014 after you affected your 2014 earnings it will still work.

Does it help?

**Comment by beledouxdenis** [comment]
Isn't this issue supposed to be reported in the Enterprise repository ?

**Comment by qdp-odoo** [comment]
yes Densi, you're probably right :) but I beleive there isn't any issue at all, actually ^^

**Comment by bistaray** [comment]
(Storing this here so I can find it later)

How do I migrate, transact through a partial year, then move into a new one with Odoo v9 Enterprise?

Let’s follow a simple scenario:
I meet with my friends and, among other things, we discuss if we can make money selling our respective services hourly.  We all agree to start a business and put in $1000 each as a loan.  There are five of us.

We create a Journal Entry in Odoo to signify the start of the Company on August 1st, 2016:

![image](https://cloud.githubusercontent.com/assets/13910162/18022454/e7508abe-6ba3-11e6-844a-e622aa1d3856.png)

On **August 1st** our Reports look like this:

![image](https://cloud.githubusercontent.com/assets/13910162/18022464/fb2d57f6-6ba3-11e6-8479-da2e73c39ca9.png)

**August 2nd:**
The next day, we enter Invoices for things we need to purchase for the business:

![image](https://cloud.githubusercontent.com/assets/13910162/18022469/068dea16-6ba4-11e6-94e1-6aec20c3c4ea.png)

Prior to registering payment for any of these Invoices, let’s review the Reports:

![image](https://cloud.githubusercontent.com/assets/13910162/18022472/0f745f02-6ba4-11e6-8157-ffc1ae102e27.png)

Notice that at the bottom, CURRENT YEAR EARNINGS.  Odoo is calculating this amount automatically for us.  Since all we have done is incur expenses, this is negative (a loss). More about this later.
Also, since we have incurred expenses of $217.50, we can see this total on the Profit and Loss Report.

**August 3rd:**
We sell some of our services and enter Customer Invoices:

![image](https://cloud.githubusercontent.com/assets/13910162/18022478/1dc60ea2-6ba4-11e6-80e3-484dc3aa03cf.png)


Prior to registering payment for any of these Invoices, let’s review the Reports:

![image](https://cloud.githubusercontent.com/assets/13910162/18022492/301f96cc-6ba4-11e6-8b0f-5fa6426c6fa2.png)


Note that CURRENT YEAR EARNINGS is updated again!

Do you see the relationship between the NET PROFIT line on the Profit and Loss Report and the CURRENT YEAR EARNINGS line on the Balance Sheet?  

On both August 2nd and August 3rd, it matches.  Odoo is calculating CURRENT YEAR EARNINGS based on what the Profit and Loss Report shows!

**August 4th.**  
We record payment from our Customers.  

![image](https://cloud.githubusercontent.com/assets/13910162/18022497/3a78109a-6ba4-11e6-8075-98c7183c4759.png)


This does not change our financial position; it just moves assets from receivables to the Bank. The Profit and Loss is the same, and the Balance Sheet just shows the movement of money moved from ‘expected’ revenue to ‘banked revenue’:

![image](https://cloud.githubusercontent.com/assets/13910162/18022501/45f07e4e-6ba4-11e6-9780-27a4b3fa2e14.png)


**August 5th**
We record payment to our Vendors.  

![image](https://cloud.githubusercontent.com/assets/13910162/18022503/4d60963c-6ba4-11e6-84ea-e82ba9980473.png)

Again, small changes. This takes care of our current liabilities and reduces our bank balance. The Reports now:

![image](https://cloud.githubusercontent.com/assets/13910162/18022519/57ba29b8-6ba4-11e6-9af4-d88ae7286c34.png)

**September**:
We sell more services, take more payments from Customers:

![image](https://cloud.githubusercontent.com/assets/13910162/18022520/5fcd3a78-6ba4-11e6-96d0-2dd565a8c63c.png)

Reports:

![image](https://cloud.githubusercontent.com/assets/13910162/18022527/6c44e558-6ba4-11e6-9dcb-827c690666c6.png)

Notice how the additional $5,500 in Revenue is added to both reports.

**October**:
Three company founders (the ones selling their services) take back their initial loan amounts.

![image](https://cloud.githubusercontent.com/assets/13910162/18022531/76cb994a-6ba4-11e6-8a97-da56a61f4a7d.png)

Reports:

![image](https://cloud.githubusercontent.com/assets/13910162/18022547/a6fd8128-6ba4-11e6-9cfc-2efa07730e44.png)


Note the Asset and Liability movement on the Balance Sheet, and no changes on the Profit and Loss.

**End of the year**.  

What do we need to do in Odoo?  We need to move the CURRENT YEAR EARNINGS into RETAINED EARNINGS:

![image](https://cloud.githubusercontent.com/assets/13910162/18022560/c411535c-6ba4-11e6-9f9e-5ba3a69c0281.png)

(Note: NOT SURE OF THE CORRECT CREDIT ACCOUNT TO USE HERE – SOMETHING FOR EQUITY?)

**Jan 1st, 2017**:

Let’s now look at the Financial Reports:

![image](https://cloud.githubusercontent.com/assets/13910162/18022569/dbc5490e-6ba4-11e6-93f4-462120dccf66.png)

Note how the Balance Sheet accounts are moved forward, and the Income and Expenses are reset to $0.

To recap: this was based on a SINGLE journal entry with two lines:

![image](https://cloud.githubusercontent.com/assets/13910162/18022576/ea9baee6-6ba4-11e6-9ad8-67b12d6ca3d1.png)

**Comment by murtadakarjikar** [comment]
I agree with the working.  In version 9, Odoo is allowing the users to pass Profit/Loss entry manually after the fiscal year is closed. Previously there was no account for current year earnings and the profit was transferred to retained earnings when opening entry was generated. 

The account 999999 (undistributed profit) is a default account and can be changed as per preferences. I would keep it the same way as sequentially the code is the last number in the COA. This account computes the Profit or loss for a particular fiscal year. Generally, no entry is to be passed to this account except for transferring the balance (profit or loss) to retained earnings. 

The snapshot shows the characteristic of this account. 
![image](https://cloud.githubusercontent.com/assets/21281223/18030356/ca2974f4-6c80-11e6-9570-ac7c4b165699.png)

Hope this helps

**Comment by gdgellatly** [comment]
The name of that account is strange - the very meaning of Retained Earnings
is Undistributed Profits, mostly, Retained Earnings can have other uses
like buybacks, and other reserve accounts could be used for distributions
(although they wouldn't be profits), but in normal parlance people
understand that Retained Earnings are the undistributed profits of a firm.

Don't see why it isn't just called Current Year Earnings.  Undistributed
simply means it has not been distributed (e.g. by dividend), a completely
different concept.

**Comment by murtadakarjikar** [comment]
The correct name of this account is Profit and Loss account. 

After year end is finalized, the balance of this account is required to be transferred to Retained Earnings as this account will start computing profitability for the new year once entries start flowing in.

**Comment by treviser** [comment]
Current Year Earnings and legal statements - My observations and conclusions - Your comments?

I would like to amend @bistaray 's write-up to make even more understandable what happens behind the scene and make it compatible with plain Odoo CE.

Observations (might be obvious, but important for the understanding):

1. The Current Year Earnings (CYE) account 999999 is not contained in the Profit and Loss statement.

If you try to set up an account of type CYE which is part of the "regular" chart of accounts and you want to post to this account, you will get an error message concerning unbalanced entries. So there is no way to include a CYE account in the (default) P&L statement.
Hence postings on the CYE account (i.e. the closing entries to carry over the current year earnings to an appropriate equity account) have no impact on the P&L. Therefore the "perpetual" P&L accumulates over time, but the same does the corresponding equity and we have an inconsistency (double counting) between the "perpetual" P&L and the "to date" Balance Sheet (from a mathematical point of view this is not nice).

2. The current year earnings entry in the balance sheet is the calculated profit/loss from the P&L statement of the same period defined for the balance sheet report.

This is true whenever you carry over the current year earnings to equity.

Conclusions

If you carry over the current year earnings to equity at the end of the current year, your balance sheet afterwards shows the current year earnings AND the augmented equity, which is wrong. In this case the balance sheet is only valid when printed before the carry over.
Therefore the carry over of the current year earnings to equity should be dated as of the first day of the following period.

It is absolutely mandatory to always specify the "Date from" and the "Date to" and not only the "Date to" date when creating a balance sheet.

Any comments, critics or confirmation are very welcome.
